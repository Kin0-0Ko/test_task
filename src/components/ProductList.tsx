import React, {FC} from 'react'
import { Product } from '../store/slieces/ProductSlice'

type ProductListProps = {
	products: Product[],
	callback: (product: Product) => void
}

const ProductList: FC<ProductListProps> = ({products, callback}) => {
  return (
	<>
		{products.map(el => 
			<tr onClick={() => callback(el)} key={el.id} style={{background: el.color, height: '30px'}}>
				<th>{el.id}</th>
				<th>{el.name}</th>
				<th>{el.year}</th>
			</tr>
			)}
	</>
  )
}

export default ProductList
