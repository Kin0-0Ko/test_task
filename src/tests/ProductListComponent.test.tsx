import React from "react";
import ProductList from '../components/ProductListComponent';
import { fireEvent, render, screen } from "@testing-library/react";

const mockObject = {
	id:1,
	name:"cerulean",
	year:2000,
	color:"#98B2D1",
	 pantone_value:"15-4020"
	}
const onCLick = jest.fn()

describe("Test ProductList", () => {

it("Snapshot product list component with props", () => {
	const {asFragment } = render(<ProductList products={[mockObject]}/>)
	expect(asFragment()).toMatchSnapshot()
})
it("should render product list component", () => {
		render(<ProductList products={[mockObject]} callback={onCLick}/>)
		expect(screen.getByTestId('product-elem')).not.toBeNull()
})
it("should not render product list component", () => {
	render(<ProductList products={[]}/>)
	expect(screen.queryByTestId('product-elem')).toBeNull()
})
it("Snapshot product list component with empty props", () => {
	const {asFragment } = render(<ProductList products={[]}/>)
	expect(asFragment()).toMatchSnapshot() 
})

it("onCLick works", () => {
	render(<ProductList products={[mockObject]} callback={onCLick}/>)
	fireEvent.click(screen.getByTestId('product-elem'))
	expect(onCLick).toHaveBeenCalled()
})

})