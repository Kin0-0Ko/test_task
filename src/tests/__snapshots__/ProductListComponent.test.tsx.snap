// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProductList Snapshot product list component 1`] = `
<DocumentFragment>
  1cerulean2000
</DocumentFragment>
`;

exports[`Test ProductList Snapshot product list component with empty props 1`] = `<DocumentFragment />`;

exports[`Test ProductList Snapshot product list component with props 1`] = `
<DocumentFragment>
  1cerulean2000
</DocumentFragment>
`;
